test/integration/scheduler/preemption/preemption_test.go

```diff
@@ -111,18 +114,39 @@ func (fp *tokenFilter) Name() string {
 
 func (fp *tokenFilter) Filter(ctx context.Context, state *framework.CycleState, pod *v1.Pod,
 	nodeInfo *framework.NodeInfo) *framework.Status {
+	if _, err := state.Read(tokenFilterName); err != nil {
+		// Should be bug.
+		// We don't store state only when PreFilter returned Skip.
+		// In other words, if reaches here, PreFilter returned Skip but somehow Filter is called.
+		return framework.NewStatus(framework.Error, err.Error())
+	}
+
 	if fp.Tokens > 0 {
 		fp.Tokens--
 		return nil
 	}
-	status := framework.Unschedulable
-	if fp.Unresolvable {
-		status = framework.UnschedulableAndUnresolvable
-	}
-	return framework.NewStatus(status, fmt.Sprintf("can't fit %v", pod.Name))
+	return fp.FilterStatus
+}
+
+type stateData struct{}
+
+func (sd *stateData) Clone() framework.StateData {
+	return sd
 }
 
 func (fp *tokenFilter) PreFilter(ctx context.Context, state *framework.CycleState, pod *v1.Pod) (*framework.PreFilterResult, *framework.Status) {
+	if pod.Name == fp.PreFilterTargetPodName && fp.PreFilterStatus.IsSkip() {
```

Comment:
I think here we should only check is: (not quite a fan to check the pod name if we can achieve this by the following logic)

```go
if fp.PreFilterStatus.Code() == framework.UnschedulableAndUnresolvable || fp.PreFilterStatus.IsSkip() {
	return nil, fp.PreFilterStatus
}
```